lib/src/lib.rs

```diff
@@ -64,6 +64,26 @@ impl QueueType {
             QueueType::VideoCaptureMplane | QueueType::VideoOutputMplane
         )
     }
+
+    /// Returns whether the queue type is output.
+    pub fn is_output(&self) -> bool {
+        matches!(
+            self,
+            QueueType::VideoOutput
+                | QueueType::VideoOutputMplane
+                | QueueType::VideoOverlay
+                | QueueType::VideoOutputOverlay
+                | QueueType::VbiOutput
+                | QueueType::SlicedVbiOutput
+                | QueueType::SdrOutput
+                | QueueType::MetaOutput
+        )
+    }
+
+    /// Returns whether the queue type is capture.
+    pub fn is_capture(&self) -> bool {
+        !self.is_output()
+    }
```

Comment:
How about turning this into a `direction()` method that return an `enum { Output, Capture }`? On top of introducing only one method, this would also allow us to use a `match` statement that lists all the variants and would trigger an error if new queue types are introduced in the future.

Reply:
Sounds good! I'll try that.